core_v2/sources/v2_domains.move

```diff
@@ -0,0 +1,1225 @@
+
+        validate_registration_duration(renewal_duration_secs);
+        assert!(is_domain_in_renewal_window(domain_name), error::invalid_state(EDOMAIN_NOT_AVAILABLE_TO_RENEW));
+        let price = v2_price_model::price_for_domain(length, renewal_duration_secs);
+        // pay the price
+        coin::transfer<AptosCoin>(sign, v2_config::fund_destination_address(), price);
+        renew_domain_internal(domain_name, renewal_duration_secs, price);
+    }
+
+    fun renew_domain_internal(
+        domain_name: String,
+        renewal_duration_secs: u64,
+        price: u64,
+    ) acquires CollectionCapability, NameRecord, RenewNameEvents, ReverseRecord {
+        let record = get_record_mut(domain_name, option::none());
+        record.expiration_time_sec = record.expiration_time_sec + renewal_duration_secs;
+
+        // Idea here is that if this is a primary name, then the target_addr's reverse lookup should point back to this domain
+        let is_primary_name = if (option::is_some(&record.target_address)) {
+            let maybe_reverse_record = get_reverse_lookup(*option::borrow(&record.target_address));
+            if (option::is_some(&maybe_reverse_record)) {
+                let reverse_record_addr = *option::borrow(&maybe_reverse_record);
+                get_token_addr_inline(domain_name, option::none()) == reverse_record_addr
+            } else {
+                false
+            }
+        } else {
+            false
+        };
+
+        // log the event
+        event::emit_event<RenewNameEvent>(
+            &mut borrow_global_mut<RenewNameEvents>(@aptos_names_v2).renew_name_events,
+            RenewNameEvent {
+                domain_name,
+                subdomain_name: option::none(),
+                renewal_fee_octas: price,
+                expiration_time_secs: record.expiration_time_sec,
+
+                target_address: record.target_address,
+                is_primary_name,
+            },
+        );
+    }
+
+    // === SUBDOMAIN MANAGEMENT ===
+
+    /// Disable or enable subdomain owner from transferring subdomain as domain owner
+    public fun set_subdomain_transferability_as_domain_owner(
+        router_signer: &signer,
+        sign: &signer,
+        domain_name: String,
+        subdomain_name: String,
+        transferrable: bool
+    ) acquires CollectionCapability, NameRecord {
+        assert!(address_of(router_signer) == @router_signer, error::permission_denied(ENOT_ROUTER));
+        validate_subdomain_registered_and_domain_owned_by_signer(sign, domain_name, subdomain_name);
+        let name_record_address = get_token_addr(domain_name, option::some(subdomain_name));
+        let transfer_ref = &borrow_global_mut<NameRecord>(name_record_address).transfer_ref;
+        if (transferrable) {
+            object::enable_ungated_transfer(transfer_ref);
+        } else {
+            object::disable_ungated_transfer(transfer_ref);
+        }
+    }
+
+    public fun transfer_subdomain_owner(
```

Comment:
I don't think we need this and it isn't being used in this contract, let's remove it. if we ever get some dispatch service in move, this will be useful.